docs/start/resourceconfig.md

```diff
@@ -0,0 +1,78 @@
+# Configure Resource Limits for Azure Pipelines Agent
+
+## Linux
+
+### Memory
+When the agent on a Linux system that is under high memory pressure, it is important to ensure that the agent does not get killed or become otherwise unusable.
+If the agent process dies or runs out of memory it cannot stream pipeline logs or report pipeline status back to the server, so it is preferable
+to reclaim system memory from pipeline job processes before the agent process.
+
+#### CGroups
+`cgroups` can be used to prevent job processes from consuming too many resources or to isolate resources between multiple agents. For a single agent it is useful
+to isolate the agent from the jobs it runs. The following `cgconfig.conf` [file](https://linux.die.net/man/5/cgconfig.conf) sets up two cgroups that impose different
+memory limits:
+
+```
+group agent {
+    memory {}
+}
+group job {
+    memory {
+        memory.limit_in_bytes = 7g;
+    }
+}
+```
+In this scenario, the agent cgroup has no limit set, so it inherits from the root or parent cgroup. By default, this would be the root cgroup which has no limit. The job cgroup
+imposes a limit of 7g of memory. This can be used in conjunction with the following `cgrules.conf` [config file](https://linux.die.net/man/5/cgrules.conf):
+
+```
+agent_user:Agent.Listener memory agent
+agent_user:Agent.Worker memory agent
+agent_user memory job
+```
+It is important to use two groups, because otherwise the pipeline job processes will inherit the group from their parent, the agent, so there will be no distinction in terms of control.
```

Comment:
Move some of this discussion up first, so we know why we're setting up 2 groups.